### ISSUE TITLE
Offline API documentation

### ISSUE THREAD

**Issue by amitguptagwl** [OpenGenus/quark#128]
**This is a(n):**

- [x] New feature proposal
- [ ] Reporting a keyword
- [ ] Error
- [ ] Proposal to the Search Engine
- [ ] Other

**Details:**
I work on many programming /scripting languages. Hence mix their syntaxe. It's more frustrating when I'm out of Internet because of xyz reason.

I wonder if we can bring API documentation of famous libraries also offline. Eg. devdocs.io , previously this site was working offline through chrome plugin but now I need to be online. If we bring some feature like them

**Would you like to work on this issue?**

- [x] Yes
- [ ] No

**Comment by amitguptagwl** [comment]
Sure, I'll start it in few days. And will update you with the progress.

**Comment by amitguptagwl** [comment]
I'm also wondering if we should keep documentation version wise. And if there is any small change then should we maintain the whole copy for another version or just the change?

It'll be little tricky to implement but can save a lot of space. In addition, it can also compare multiple versions for changes.

**Comment by AdiChat** [comment]
Yes, for different versions of the same documentation, we must store only the difference. This will improve the setup time and will result in great features like comparing different versions, analysing progress of various technologies and others. :+1: 

For most basic users, the latest version of a documentation should address the majority of the queries. Different versions will be useful for advanced or experienced users. With the development of additional features, it will be of interest to all users. :smile: 

You may, initially, work on integrating the latest (or any one) version of a particular documentation. :+1: 

Once done, we can work on the feature of storing different versions as a separate feature. This will be globally applicable as we can use the same structure for the documentations, code data-store and other future data-related feature. :sparkles:

**Comment by amitguptagwl** [comment]
I've just checked and found that [dedocs](https://github.com/freeCodeCamp/devdocs) is already open source and works offline once cloned locally. So I believe there is no need to create another repo.

Going further, I've created a [collection](https://github.com/amitguptagwl/offline) for such tools.

**Comment by AdiChat** [comment]
Okay :+1: 

Our approach towards solving this problem is more general and goes beyond documentation.
If anyone wants to work on this feature, free feel to let us know and go for it. :smile:

**Comment by amitguptagwl** [comment]
I believe you mean version difference. What should be the end result in that case? As the process will be automated, what we can show is just text difference. We can't actually say which api is added, removed, or updated.

let me know what do you think.
